Organizer custom notification message

Erza-T opened the pull request:
Closes #32, #33, #34, and #25.

To test,:

- You need at least 2 devices. One for organizer and one for entrant.
-You can either follow the steps for the entrant to join the organizers event, or directly add your entrant device id to the lists (waitlist, enrolled, cancelled, invited) in your Firestore events document. 

- Go to the notifications activity the hosted events screen, and input your custom notification. Make sure to check off the type of entrant you want to send the notification to.
- Your entrant device should see the custom notification, and the notification document should be created.

Erza-T committed:
Organizer can now create a custom notificaiton message, minor bug fixes

sgctcub commented:
- custom notification has notification details field that doesn't appear on the notification for me. Is that intended? if it is what's the point of that field
- there is a small typo in invited notification, no space between event and event name
- the selected entrants notification did not work. i added myself to all 5 lists in the event.
- i checked all the boxes and sent a custom notification, should i be getting the same notification 4 times? i just got it once

Erza-T committed:
selected notification bug fix, invited toast message bug fix

Erza-T committed:
from main

Erza-T commented:
- If you're talking about the extra notification details, this is notification details you will look at if the user clicks on the down arrow from the notification bar
- fixed typo
- fixed selected entrants notification
- if a user is part of multiple checkboxes, it should only send the notification once (it's unusual to be sent the same exact notification to the same user at the same time
